internal/storage/storage_handle_test.go

```diff
@@ -253,6 +254,186 @@ func (testSuite *StorageHandleTest) TestCreateHTTPClientHandle_WithGRPCClientPro
 	assert.Contains(testSuite.T(), err.Error(), fmt.Sprintf("client-protocol requested is not HTTP1 or HTTP2: %s", cfg.GRPC))
 }
 
+func (testSuite *StorageHandleTest) TestCreateHTTPClientHandle_WithReadStallRetry() {
+	testCases := []struct {
+		name                 string
+		enableReadStallRetry bool
+	}{
+		{
+			name:                 "ReadStallRetryEnabled",
+			enableReadStallRetry: true,
+		},
+		{
+			name:                 "ReadStallRetryDisabled",
+			enableReadStallRetry: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		testSuite.Run(tc.name, func() {
+			sc := storageutil.GetDefaultStorageClientConfig()
+			sc.ReadStallRetryConfig.Enable = tc.enableReadStallRetry
+
+			storageClient, err := createHTTPClientHandle(context.Background(), &sc)
```

Comment:
Is there any way to test that the returned storageClient has read-stall-retry enabled/disabled as expected ?

Reply:
Yes, there is a way to do this, but that requires integration with storage-test-bench server (similar to retry error).